analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd

```diff
@@ -0,0 +1,318 @@
+
+The purpose is to identify samples that may need to be reclassified as LGAT instead of HGAT using t-SNE and UMAP clustering.
+
+## Usage
+
+This notebook is intended to be run via the command line from the top directory
+of the repository as follows:
+
+```
+Rscript -e "rmarkdown::render('analyses/molecular-subtyping-HGG/09-HGG-with-braf-clustering.Rmd', clean = TRUE)"
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+# Load in tidyverse functions
+library(tidyverse)
+```
+
+### Directories
+
+```{r}
+# File path to data directory
+data_dir <- file.path("..", "..", "data")
+
+# File path to plots directory -- this contains the multipanel output plots
+plots_dir <- file.path("plots")
+
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+```
+
+### Functions
+
+```{r}
+# Source the functions for plotting dimension reduction scores
+source(
+  file.path(
+    "..",
+    "transcriptomic-dimension-reduction",
+    "util",
+    "dimension-reduction-functions.R"
+  )
+)
+```
+
+#### Function to join dimension reductions scores with HGG molecular subtype file
+
+```{r}
+align_data <- function(aligned_scores_df,
+                       hgg_molecular_subtype_df) {
+  # Given a data.frame with dimension reduction scores already aligned with the
+  # metadata, join the `molecular_subtype` column from the 
+  # `HGG_molecular_subtype.tsv` file generated in this module.
+  # Args:
+  #   aligned_scores_df: data.frame containing dimension reduction scores
+  #                      and relative information from the metadata
+  #   hgg_molecular_subtype_df: data.frame containing a `molecular_subtype`
+  #                             column with values determined in this module
+  # Returns:
+  #   aligned_scores_df: data.frame containing dimension reduction scores,
+  #                      metadata, and the `molecular_subtype` column from the
+  #                      `results/HGG_molecular_subtype.tsv` file
+  
+  aligned_scores_df <- aligned_scores_df %>%
+    dplyr::left_join(
+      hgg_molecular_subtype_df,
+      by = c("Kids_First_Biospecimen_ID" = "Kids_First_Biospecimen_ID_RNA")
+    ) %>%
+    dplyr::filter(
+      broad_histology %in% c(
+        "Low-grade astrocytic tumor",
+        "Diffuse astrocytic and oligodendroglial tumor"
+      )
+    ) %>%
+    dplyr::group_by(Kids_First_Biospecimen_ID) %>%
+    dplyr::select(-molecular_subtype.x, molecular_subtype = molecular_subtype.y) %>%
+    dplyr::distinct() %>%
+    dplyr::mutate(
+      subtype_class = dplyr::case_when(
+        stringr::str_detect(molecular_subtype, "BRAF V600E") &
+          broad_histology == "Diffuse astrocytic and oligodendroglial tumor" ~ "HGAT_BRAF",
+        broad_histology == "Low-grade astrocytic tumor" ~ "LGAT",
+        TRUE ~ "HGAT_NO_BRAF"
+      )
+    )
+  
+  return(aligned_scores_df)
+}
+```
+
+#### Function to generate a multipanel plot
+
+```{r}
+generate_multipanel_plot <- function(plot1,
```

Comment:
If we open a new pull request to rerun `transcriptomic-dimension-reduction`, maybe we can add a new file to `util` in that module that contains this function that can be sourced here and in `analyses/transcriptomic-dimension-reduction/scripts/generate-multipanel-plot.R`?